### DISCUSSION TITLE
How can I use a NaiveDate field type?

### DISCUSSION THREAD

**Discussion by joshsharp** [weldsorm/welds#42] [Q&A] [answered]
Hi! Thanks for making Welds, I find the ergonomics the nicest of any Rust ORM I've tried :)

I'm trying to make a model that has a `chrono::NaiveDate` as field type, and I'm using sqlite. I know sqlx supports this, and I can see that migrations support making a date type field that maps to a `NaiveDate`, but when I try to define my struct, the `WeldsModel` macro complains. E.g.:

```rust
#[derive(Debug, WeldsModel, Clone, PartialEq)]
#[welds(table = "track")]
pub struct Track {
    #[welds(primary_key)]
    pub id: i64,
    pub filename: String,
   // and so on...
    pub date_added: NaiveDate,
    pub last_played: Option<NaiveDate>,
}
```

The error is as you'd expect: "the trait bound `NaiveDate: Param` is not satisfied".

Is there some way I can get around this without implementing this trait myself? I'd rather not.

Thanks!

**Comment by lex148** [comment, marked as answer]
Hey @joshsharp so glad to hear you are enjoying welds. I need to improve the documentation here. It really isn't clear what to do when dealing with chrono/uuid/serde types. If you are using types that require manually enabling in sqlx (mysql,sqlite,portgres) that needs to be done in the cargo.toml file. Specifically you need to enable the sqlx feature. 

Here Is an example of what I think you are looking for. 

Cargo.toml
```rust
[package]
name = "welds_example_dates"
version = "0.1.0"
edition = "2021"

[dependencies]
chrono = "0.4.38"
sqlx = { version = "0.8.2", features = ["chrono", "runtime-tokio", "sqlite"] }
tokio = { version = "1.41.1", features = ["full"] }
welds = { version = "0.4.5", features = ["migrations", "sqlite"] }
```

src/main.rs
```rust
use welds::prelude::*;

#[tokio::main]
async fn main() -> welds::errors::Result<()> {
    // connect to database, this will be different for you im sure
    let connection_string = "sqlite::memory:";
    let client = welds::connections::connect(connection_string).await?;

    // create table for example
    client.execute("CREATE TABLE example (id INTEGER PRIMARY KEY AUTOINCREMENT,date_col DATE,time_col TIME,datetime_col DATETIME);", &[]).await?;

    Example::new().save(client.as_ref()).await?;
    Example::new().save(client.as_ref()).await?;

    let count = Example::all().count(client.as_ref()).await?;
    println!("Total Rows: {}", count);
    Ok(())
}

#[derive(WeldsModel, Clone, PartialEq, Debug, Default)]
#[welds(table = "example")]
pub struct Example {
    #[welds(primary_key)]
    pub id: i64,
    pub date_col: chrono::NaiveDate,
    pub time_col: chrono::NaiveTime,
    pub datetime_col: chrono::NaiveDateTime,
}
```

The migrations for time types are in the latest on git, but be aware they haven't hit the latest release yet. That should be out here in the next couple of days.

**Reply by joshsharp** [reply]
Thanks so much, that did the trick :) Appreciate the quick update to the package and the readme too.
